fabric_cf/actor/core/policy/network_node_inventory.py

```diff
@@ -247,9 +248,14 @@ def allocate(self, *, reservation: ABCReservationMixin, graph_id: str, graph_nod
             raise BrokerException(error_code=Constants.INVALID_ARGUMENT,
                                   msg=f"resource type: {requested.get_type()}")
 
+        # Always use requested capacities to be mapped from flavor i.e. capacity hints
+        requested_capacity_hints = requested.get_capacity_hints()
```

Comment:
I think this needs to be more careful - user can choose to set capacities OR capacity_hints. If capacities are set, they need to be mapped onto a instance type and capacity hint should be set, if capacity hint is set, then its capacity should be set to capacities. I could be missing something.

Reply:
This mapping is done in orchestrator_slice_wrapper.py OrchestratorSliceWrapper::create. So when a request reaches Broker or AM, capacity hints are always set.

Orchestrator gives preference to Capacity Hints if both are set and ignores Capacities. If Capacity Hints are not set, Capacity Hints are computed.

Reply:
https://github.com/fabric-testbed/ControlFramework/pull/91/files?file-filters%5B%5D=.py&file-filters%5B%5D=.txt#diff-d3fd402837b4319120094494770a24f548ec47483ae1f0aeef10fde52bb9e6dd

Reply:
Right. What happens if neither is set (which is an error, to be sure). Should we check for it somewhere earlier?

Reply:
I assumed, experiment topology or build sliver would fail if neither is set. I can explicitly add those checks to validate incoming slivers.

Reply:
I don't think we can trust anything coming over the wire. There are two ways to check - procedurally in your code or using cypher to add extra rules just to check slice validity (we have rules to check general model validity, but slices have more rules).